### ISSUE TITLE
SONIC_PreData  Please download the model first

### ISSUE THREAD

**Issue by ohnitescent** [smthemex/ComfyUI_Sonic#60]
Hi Dear team ;-)
I'm using Colab (A100) and I've this error : 
SONIC_PreData - Please download the model first
__________________________________________________________
Everything was installed following your guidance (I've checked, all the files are in there folder. There's no missing nodes also (checked via Manager)
Can you help please ? Many tx in advance.
__________________________________________________________

# ComfyUI Error Report
## Error Details
- **Node ID:** 3
- **Node Type:** SONIC_PreData
- **Exception Type:** Exception
- **Exception Message:** Please download the model first
## Stack Trace
```
  File "/content/drive/MyDrive/ComfyUI/execution.py", line 327, in execute
    output_data, output_ui, has_subgraph = get_output_data(obj, input_data_all, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)
                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/content/drive/MyDrive/ComfyUI/execution.py", line 202, in get_output_data
    return_values = _map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/content/drive/MyDrive/ComfyUI/execution.py", line 174, in _map_node_over_list
    process_inputs(input_dict, i)

  File "/content/drive/MyDrive/ComfyUI/execution.py", line 163, in process_inputs
    results.append(getattr(obj, func)(**inputs))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/content/drive/MyDrive/ComfyUI/custom_nodes/ComfyUI_Sonic/sonic_node.py", line 141, in sampler_main
    raise Exception("Please download the model first")

**Comment by smthemex** [comment]
some version of comfyUI can't use customize path ,you can edit [here](https://github.com/smthemex/ComfyUI_Sonic/blob/main/sonic_node.py#L135)  and [here](https://github.com/smthemex/ComfyUI_Sonic/blob/main/sonic_node.py#L136) to absolute path, will avoid this inspection.

**Comment by ohnitescent** [comment]
Dear smthemex. 1st thank you for your answer. I tried ;-( I don't want to bother you, just to get the model working ;-)
1) I am using the official file :  Comfyui_colab_with_manager.ipynb
2) I modified the sonic_node.py file (copy below). 
3) I first got a message, even though it was installed correctly (after checking custom nodes, templates, checkpoints), that the files were missing: SONICSampler, SONICLoader, SONIC-PreData
I had to uninstall and reinstall via Manager to not see the message anymore. I did a consistency check to see that it was installed correctly. It was.
3) I re-ran the model (everything was in their right folders). But I got the same message as before "SONIC-PreDataPlease download the model first"
4) I copied again the sonic_node.py in his folder (custom_nodes/ComfyUI_Sonic
5) Did a run : same message "SONIC_PreData - Please download the model first"

a) Can you please check the sonic_node.py modified according to your advice?
b) Which version of comyui plus manager are you using please ?

Many thanks in advance for your answer and time !

=============
#!/usr/bin/env python
# -*- coding: UTF-8 -*-
import os
import torch
import gc
import numpy as np
from omegaconf import OmegaConf
from diffusers import AutoencoderKLTemporalDecoder
from diffusers.schedulers import EulerDiscreteScheduler
from transformers import WhisperModel, AutoFeatureExtractor
import random
import io
import torchaudio
from .src.models.base.unet_spatio_temporal_condition import UNetSpatioTemporalConditionModel
from .sonic import Sonic, sonic_predata, preprocess_face, crop_face_image
from .src.dataset.test_preprocess import image_audio_to_tensor
from .src.models.audio_adapter.audio_proj import AudioProjModel
from .src.models.audio_adapter.audio_to_bucket import Audio2bucketModel
from .node_utils import tensor2cv, cv2pil, convert_cf2diffuser, tensor_upscale, tensor2pil
from .src.dataset.face_align.align import AlignImage

import folder_paths

MAX_SEED = np.iinfo(np.int32).max
current_node_path = os.path.dirname(os.path.abspath(__file__))

device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")

# Ensure custom model path is correctly set
SONIC_weights_path = os.path.join(folder_paths.models_dir, "sonic")
if not os.path.exists(SONIC_weights_path):
    os.makedirs(SONIC_weights_path)
folder_paths.add_model_folder_path("sonic", SONIC_weights_path)

class SONICLoader:
    def __init__(self):
        pass

    @classmethod
    def INPUT_TYPES(s):
        return {
            "required": {
                "model": ("MODEL",),
                "sonic_unet": (["none"] + folder_paths.get_filename_list("sonic"),),
                "ip_audio_scale": ("FLOAT", {"default": 1.0, "min": 0.5, "max": 2.0, "step": 0.1}),
                "use_interframe": ("BOOLEAN", {"default": True}),
                "dtype": (["fp16", "fp32", "bf16"],),
            },
        }

    RETURN_TYPES = ("MODEL_SONIC", "DTYPE")
    RETURN_NAMES = ("model", "dtype")
    FUNCTION = "loader_main"
    CATEGORY = "SONIC"

    def loader_main(self, model, sonic_unet, ip_audio_scale, use_interframe, dtype):
        weight_dtype = torch.float16 if dtype == "fp16" else torch.float32 if dtype == "fp32" else torch.bfloat16
        
        svd_repo = os.path.join(current_node_path, "svd_repo")
        flownet_ckpt = os.path.join(SONIC_weights_path, "RIFE")
        
        if sonic_unet != "none":
            sonic_unet = folder_paths.get_full_path("sonic", sonic_unet)

        val_noise_scheduler = EulerDiscreteScheduler.from_pretrained(svd_repo, subfolder="scheduler")
        
        unet_config_file = os.path.join(svd_repo, "unet")
        unet = convert_cf2diffuser(model.model, unet_config_file, weight_dtype)
        
        vae_config = os.path.join(svd_repo, "vae/config.json")
        vae_config = OmegaConf.load(vae_config)

        pipe = Sonic(device, weight_dtype, vae_config, val_noise_scheduler, unet, flownet_ckpt, sonic_unet,
                     use_interframe, ip_audio_scale)

        gc.collect()
        torch.cuda.empty_cache()
        return (pipe, weight_dtype)


NODE_CLASS_MAPPINGS = {
    "SONICLoader": SONICLoader,
}

NODE_DISPLAY_NAME_MAPPINGS = {
    "SONICLoader": "SONICLoader",
}

**Comment by ohnitescent** [comment]
Dear smthemex, finally I arrived to solve my issue. I don't disturb you anymore. Again thanks for your quick answer. And thanks for your work !! Take care ;-) Best regards. ON.

**Comment by ohnitescent** [comment]
Dear smthemex, finally I arrived to solve my issue. I don't disturb you anymore. Again thanks for your quick answer. And thanks for your work !! Take care ;-) Best regards. ON.
